packages/esbuild-plugin-node/src/config/main.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  InstrumentationBase,
+  InstrumentationModuleDefinition,
+} from '@opentelemetry/instrumentation';
+
+import { EsbuildInstrumentationConfigMap } from '../types';
+import { getNodeAutoInstrumentations } from '@opentelemetry/auto-instrumentations-node';
+
+export const instrumentations: InstrumentationModuleDefinition<any>[] =
+  getNodeAutoInstrumentations()
+    .flatMap(i => {
+      // This is because the GrpcInstrumentation does not extend InstrumentationBase
+      // Should we instead have it extend that?
+      if (i instanceof InstrumentationBase) {
+        return i.init() ?? [];
+      }
+      return [];
+    })
+    .filter(Boolean);
+
+function configGenerator<T extends { enabled?: boolean }>(
+  config?: T
+): string | undefined {
+  if (!config) return;
+  return JSON.stringify(
+    Object.fromEntries(
+      Object.entries(config).filter(([, v]) => typeof v !== 'function')
```

Comment:
Since the plugin has to return a new string for the module contents in the esbuild plugin the only approach I could think of for functions was to call `.toString()` on them, which will return the function itself as a valid string that could be evaluated, but it has too many gotcha's in my opinion. The most significant of which is that if the function relies on any state from enclosing scope (imports, variables from the surrounding module) it wouldn't work